include/grpc/event_engine/event_engine.h

```diff
@@ -363,37 +368,26 @@ class EventEngine : public std::enable_shared_from_this<EventEngine> {
     /// \a default_port may be a non-numeric named service port, and will only
     /// be used if \a address does not already contain a port component.
     ///
-    /// When the lookup is complete, the \a on_resolve callback will be invoked
-    /// with a status indicating the success or failure of the lookup.
-    /// Implementations should pass the appropriate statuses to the callback.
-    /// For example, callbacks might expect to receive DEADLINE_EXCEEDED or
+    /// When the lookup is complete or cancelled, the \a on_resolve callback
+    /// will be invoked with a status indicating the success or failure of the
+    /// lookup. Implementations should pass the appropriate statuses to the
+    /// callback. For example, callbacks might expect to receive CANCELLED or
     /// NOT_FOUND.
-    ///
-    /// If cancelled, \a on_resolve will not be executed.
-    virtual LookupTaskHandle LookupHostname(LookupHostnameCallback on_resolve,
```

Comment:
Github UI won't let me comment on the relevant lines, but the `LookupTaskHandle` type is now unused and should be deleted. Same for the `LookupTaskHandleSet` and the overloads for handle-to-string conversions. Do you plan to do that cleanup in a separate step, and is so, what's the reasoning?

Reply:
The `LookupTaskHandle` type is still used in the iomgr DNSResolver interface: https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/resolve_address.h#L46-L47

The `LookupTaskHandleSet` type is used in the ares implementation of that interface (aka `AresDNSResolver`): https://github.com/grpc/grpc/blob/master/src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc#L709

Looks like `HandleToString` is also part of the iomgr DNSResolver interface. I think we can remove these when we remove the iomgr DNSResolver?

Reply:
Can we move the `LookupTaskHandle` type to `grpc_core::DNSResolver`? Doing it now allows us to make fewer, more self-contained public API changes, and we get to make them earlier, which are both wins for engine implementers.

Reply:
Makes sense. I moved `LookupTaskHandle` and `LookupTaskHandleSet` to `grpc_core::DNSResolver`. PTAL. Thanks!

Reply:
Looks great, thanks!